### PR TITLE
(Done) Update Electron

### PR DESCRIPTION
**Pull request by z11h**
This PR is going to serve as a work in progress for updating Electron to the latest version (0.37.2) from version 0.34.1. 

Electron v37 is now updated to use Chrome 49, so it comes with many stability, security and speed improvements as well as a boat load of Bug Fixes.

Checklist:
- [x] Update Dependences
- [x] Make sure the app builds + fix all bugs that arise
- [x] Clean up

:smile_cat:

### CODE REVIEW

File: src/client/settings/templates/settings-cache.html [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
         <md-input-container>
             <md-checkbox ng-model="settings.config.deleteTmpOnClose">
-                Clear Tmp Folder after closing app?
```
Comment (vankasteelj) [inline src/client/settings/templates/settings-cache.html:17]: this kind of modification, although it makes perfect sense, is also breaking i18n. We currently have most of the strings translated into 50+ langs, and changing it bits by bits will make it hard to maintain, I suggest you to keep String modifications for a bigger, condensed commit, with i18n modifications as well

Comment (z11h) [inline src/client/settings/templates/settings-cache.html:17]: alright, Ill fix it! ;)

Comment (z11h) [inline src/client/settings/templates/settings-cache.html:17]: So basically whenever I update a string I have to also update all the translations right :smile:

Comment (vankasteelj) [inline src/client/settings/templates/settings-cache.html:17]: translations used to be externalized, it really is best not to tamper with strings. You may add new ones (although I'm not sure why you would?), but modifying or rephrasing isn't a good idea at this point, I believe.

Comment (hadim) [inline src/client/settings/templates/settings-cache.html:17]: I don't agree with this. Since `butter-desktop-angular` is a complete rewrite from zero, I think we should give to devs as much as flexibility they want and they need.

Since this project is really not ready to use by lambda users, please hack it, rewrite it completely and we will care about translations later (we always find people to translate in their native languages, I am not afraid by that).

(that's only my opinion, if a third person could give his opinion that would be great)

Comment (xaiki) [inline src/client/settings/templates/settings-cache.html:17]: as i generally agree with you @hadim on the state of the project and it's hackability, i don't think it's good practice to change strings in a functional commit.

Comment (hadim) [inline src/client/settings/templates/settings-cache.html:17]: If you talking about translations in commits I agree that it should not be inside commits (maybe a separate repo ?). But for english strings inside the code I don't see why not.

Comment (z11h) [inline src/client/settings/templates/settings-cache.html:17]: @xaiki btw, my email is zridouh@live.com for the slack :)

I agree with @hadim about the hackability of the project, and I don't think that the app should break should break from the lack of translation. Also, wouldn't it be easier for people to contribute and for simplicities sake to have it inside a separate repo that we can just pull into all the butter projects.

Comment (xaiki) [inline src/client/settings/templates/settings-cache.html:17]: @hadim i'm talking about separating string changes from functional changes. there is no reason to change strings in a PR called 'update electron'.

Comment (z11h) [inline src/client/settings/templates/settings-cache.html:17]: @xaiki That is fully a mistake on my end, I forgot to switch branches when making these changes (I was still pushing to patch-2) when I was meaning to submit to different PR with these changes :smile: 

Sorry for any inconveniences that have been caused

**Comment by hadim** [comment]
Good to merge @xaiki ?
@ZakariaRidouh you should rebase this before I can merge it.

File: src/client/settings/templates/settings-cache.html (hunk 1 above)
Comment (xaiki) [inline src/client/settings/templates/settings-cache.html:17]: this is still not fixed

File: src/client/settings/templates/settings-buttons.html [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <section class="buttons" layout="row" layout-sm="column" layout-align="center center">
-    <md-button class="md-raised md-accent" ng-if="settings.config.showAdvancedSettings">Flush bookmarks database</md-button>
-    <md-button class="md-raised md-accent" ng-if="settings.config.showAdvancedSettings">Flush subtitles cache</md-button>
-    <md-button class="md-raised md-accent">Flush all databases</md-button>
-    <md-button class="md-raised md-accent">Reset to Default Settings</md-button>
+    <md-button class="md-raised md-accent" ng-if="settings.config.showAdvancedSettings"><i class="material-icons">delete_forever</i> Flush bookmarks database</md-button>
+    <md-button class="md-raised md-accent" ng-if="settings.config.showAdvancedSettings"><i class="material-icons">delete_forever</i> Flush subtitles cache</md-button>
+    <md-button class="md-raised md-accent"><i class="material-icons">delete_forever</i> Flush all databases</md-button>
+    <md-button class="md-raised md-accent"><i class="material-icons">restore</i> Reset to Default Settings</md-button>
 </section>
```
Comment (xaiki) [inline src/client/settings/templates/settings-buttons.html:6]: this should go into another PR

**Comment by xaiki** [comment]
lots of material icons changes in this PR looks like it's not needed to update electron.
